lib/gollum-lib/blob_entry.rb

```diff
@@ -47,10 +47,7 @@ def blob(repo)
     #
     # Returns a Gollum::Page instance.
     def page(wiki, commit)
-      blob         = self.blob(wiki.repo)
-      page         = ::Gollum::Page.new(wiki).populate(blob, self.dir)
-      page.version = commit
-      page
+      ::Gollum::Page.new(wiki, self.blob(wiki.repo), self.dir, commit)
     end
 
     # Gets a File instance for this blob.
```

Comment:
Just for my understanding: how can one tell if one should call `page` or `file` on a `blob_entry` object?

Reply:
I think only based on whether it has a valid page extension. I think these methods on BlobEntry are now only being called from somewhere within wiki.rb. They don't seem essential, but I decided to leave them in for convenience. We could also try to get rid of them entirely if you think that would be less confusing

Reply:
What would happen if one were to call `page` on a BlobEntry object that is actually not a page?

Reply:
I think you would end up with a Page object that would throw errors (hopefully) when you try to access the rendered_data method, because no renderer exists for the file type in question

Reply:
> I think only based on whether it has a valid page extension

Wouldn't it be preferable to hide this logic from the user? If there is no use case for trying to cast a BlobEntry into a Page while it is actually a File (I can't think of one), then why not let Gollum handle this internally? For example, `BlobEntry#contents` could return either a Gollum::Page (which is a subclass of Gollum::File) or a Gollum::File, depending on which one is fitting. Thoughts?

Reply:
Yes I was wondering about that too. We could have `to_page_or_file` method (better name?), but in all contexts where we'd use that, we actually already know whether it's going to be a page or file. [Here](https://github.com/gollum/gollum-lib/blob/gollum-lib-5.x/lib/gollum-lib/wiki.rb#L683-L684), for example, we'd still have to do the `valid_page_name?` check to exclude subpages (which have a valid page extension, but are not a valid standalone page). Do you still think it would be better to have the generic method?

Reply:
Okay. Leave it as is for now. 👍